lua/cmake-tools/init.lua

```diff
@@ -197,6 +209,15 @@ function cmake.generate(opt, callback)
     end)
   end
 
+  -- if exists cmake-kits.json, kit is used to set
+  -- environmental variables and args.
+  local kits_config = kits.parse(const.cmake_kits_path, config.cwd)
+  if kits_config and not config.kit then
+    return cmake.select_kit(function()
+      cmake.generate(opt, callback)
+    end)
+  end
+
```

Comment:
Why should move this part here?

Reply:
In the `select_build_type()` funtion -> `vim.ui.select()`[line 905], the `cmake_generate()` or the `callback()` branches do not pick-up the changes from the CMakeKits.json and do not detect the modified envScript, if you have modified the `.json` file and then re-run /re-opened vim.

Moving this before the generate callback makes sure that this does not happen, or atleast that is my idea. This is not thoroughly tested, as I was only working with `:CMakeGenerate` each time to test this feature, and not with other startup commands, like `:CMakeRun` or `:CMakeBuild`.

PS: Although there is a cmake.clean() command before this(line 115), I don't think running the envScript for clean is necessary, otherwise, it would be best to move it there.

Reply:
I don't quite understand. Why does kits have relationship with build type? Can you give an example?

Reply:
Sorry for the change, it has nothing to with build_type, ... was a mistake on my end... reverting change

## TL;DR
When I was developing the envScript thing, on my working branch I had not yet created a save method in sesions to save the envScript. So, when doing `:CmakeGenerate` on a directory which did not (yet) have a sessions file in the cache, the envScript was loaded and ran correctly. 

However, upon modifying the `CMakeKits.json` to remove(or modify) the envSvript, and reopening nvim the same directory, the changes in the file were not being detected from the modified `CMakeKits.json`, so in frustration, I moved this above the `select_build_type()` function, which solved this problem (don't exactly know how... probably had something to do with the saving of the sessions file).

Now after reverting the change and re-testing this, the bug is ... gone :D

Reply:
Interesting, lol.